File: ios/vendored/sdk49/@stripe/stripe-react-native/ios/CardFieldManager.m
#import <Foundation/Foundation.h>
#import <ABI49_0_0React/ABI49_0_0RCTBridgeModule.h>
#import <ABI49_0_0React/ABI49_0_0RCTViewManager.h>

@interface ABI49_0_0RCT_EXTERN_REMAP_MODULE(CardFieldManager, ABI49_0_0CardFieldManager, ABI49_0_0RCTViewManager)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(postalCodeEnabled, BOOL)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(countryCode, NSString)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(onCardChange, ABI49_0_0RCTDirectEventBlock)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(onFocusChange, ABI49_0_0RCTDirectEventBlock)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(cardStyle, NSDictionary)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(placeholders, NSDictionary)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(autofocus, BOOL)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(disabled, BOOL)
ABI49_0_0RCT_EXPORT_VIEW_PROPERTY(dangerouslyGetFullCardDetails, BOOL)
ABI49_0_0RCT_EXTERN_METHOD(focus:(nonnull NSNumber*) ABI49_0_0ReactTag)
ABI49_0_0RCT_EXTERN_METHOD(blur:(nonnull NSNumber*) ABI49_0_0ReactTag)
ABI49_0_0RCT_EXTERN_METHOD(clear:(nonnull NSNumber*) ABI49_0_0ReactTag)
@end

File: ios/vendored/sdk49/@stripe/stripe-react-native/ios/CardFieldManager.swift
import Foundation

@objc(ABI49_0_0CardFieldManager)
class CardFieldManager: ABI49_0_0RCTViewManager {
    override func view() -> UIView! {
        let cardField = CardFieldView()
        let stripeSdk = bridge.module(forName: "StripeSdk") as? StripeSdk
        stripeSdk?.cardFieldView = cardField;
        return cardField
    }
    
    @objc func focus(_ abi49_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI49_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFieldView = (viewRegistry![abi49_0_0ReactTag] as? CardFieldView)!
            view.focus()
        }
    }
    
    @objc func blur(_ abi49_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI49_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFieldView = (viewRegistry![abi49_0_0ReactTag] as? CardFieldView)!
            view.blur()
        }
    }
    
    @objc func clear(_ abi49_0_0ReactTag: NSNumber) {
        self.bridge!.uiManager.addUIBlock { (_: ABI49_0_0RCTUIManager?, viewRegistry: [NSNumber: UIView]?) in
            let view: CardFieldView = (viewRegistry![abi49_0_0ReactTag] as? CardFieldView)!
            view.clear()
        }
    }
    
    override class func requiresMainQueueSetup() -> Bool {
        return false
    }
}
